### ISSUE TITLE
How do you learn something new (Like React) ? 

### ISSUE THREAD

**Issue by MaraAlexa** [IonicaBizau/ama#4]
In a previous question you said you did not really use frameworks and rely on libraries but that you indeed learned React. How do you go about learning a new thing? What is the first thing you do? Do you start by reading the documentation? Or start coding a simple example which you then extend? Do you draw how the code connects through different files? Do you vizualise mentally how the code executes? Some other technique perhaps?
I am trying to learn React so how you learned React would be a good example for me.
Thanks

**Comment by IonicaBizau** [comment]
>  What is the first thing you do?

I try to summarize what I want to build.

> Do you start by reading the documentation? Or start coding a simple example which you then extend?

I quickly search examples built by other people (preferably examples from official docs, falling back to StackOverflow). :mag: 

In fact I want to have a working example I can play with and transform it to reach my goal. :rocket: 

When I don't know how to build the things, I generally ask questions on StackOverflow following these three steps:

 1. Write the question title (exactly how I'm going to write it on SO) in Google's text input and search it.
 2. If I find my answer, I stop here. Otherwise, I copy-paste it [in the question title](http://stackoverflow.com/questions/ask).
 3. If there is a similar question in the suggestions, I stop. Otherwise I write my question, including code I tried and hoping that somebody will help out. :pray: 
 4. Optionally, I play piano waiting for the answer. :musical_keyboard: 

> Do you draw how the code connects through different files? Do you vizualise mentally how the code executes?

Sometimes. I quickly make a drawing on a piece of paper. :memo: Something that nobody except me can understand. :grin:

It's hard to make a dump of what I think, but I really feel what's the code does: step by step, like in a debugger. When I'm wrong I run the code in the debug mode. :smile: 

> I am trying to learn React so how you learned React would be a good example for me.

Basically, I helped somebody having a React app set up already. The React code is pretty human readable and I quickly understood the way how I should write code. :thought_balloon:
